### PR TITLE
chore: fix findbugs errors

### PR DESCRIPTION
**Pull request by big-andy-coates**
### Description 

Fix four findbugs (and clean up code a little)

question is... why is our build not failing, given we have findbugs errors?  Could it be that spotbugs is missing this check?

Update:

Ran `mvn spotbugs:check` locally without this fix and spotbugs did not report any issues... which is bad!

Here's an example of the error in question:

```java
    throw new KsqlException(String.format(
            "Invalid CommandId string (line %d): %s",
            n, new String(record.getLeft(), StandardCharsets.UTF_8), e
        ));
```

Note how `e` is unintentionally being passed as a third parameter to `String.format` rather than as a second parameter to the exception constructor.

Findbugs does find this error.

### CODE REVIEW

File: ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/CommandTopicBackupImpl.java [hunk 1]
```diff
@@ -46,18 +46,13 @@
...
  */
 public class CommandTopicBackupImpl implements CommandTopicBackup {
+
   private static final Logger LOG = LoggerFactory.getLogger(CommandTopicBackupImpl.class);
-  private static final Ticker CURRENT_MILLIS_TICKER = new Ticker() {
-    @Override
-    public long read() {
-      return System.currentTimeMillis();
-    }
-  };
```
Comment (big-andy-coates) [inline ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/CommandTopicBackupImpl.java:55]: No need to use `Ticker` here. 

a) `Ticker` is marked as unstable
b) `Ticker.read()`'s java docs say it returns nanoseconds, not milliseconds as this impl does
c) A simple `Supplier<Long>` does what is needed.

File: ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/CommandTopicBackupImpl.java [hunk 2]
```diff
@@ -133,23 +129,22 @@ private boolean isRecordInLatestReplay(final ConsumerRecord<byte[], byte[]> reco
...
-  private void throwIfInvalidRecord(final ConsumerRecord<byte[], byte[]> record) {
+  private static void throwIfInvalidRecord(final ConsumerRecord<byte[], byte[]> record) {
     try {
       InternalTopicSerdes.deserializer(CommandId.class).deserialize(record.topic(), record.key());
     } catch (final Exception e) {
       throw new KsqlException(String.format(
           "Failed to backup record because it cannot deserialize key: %s",
-          new String(record.key(), StandardCharsets.UTF_8), e
-      ));
+          new String(record.key(), StandardCharsets.UTF_8)), e);
```
Comment (big-andy-coates) [inline ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/CommandTopicBackupImpl.java:138]: Findbugs error no 1.

File: ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/CommandTopicBackupImpl.java [hunk 3]
```diff
@@ -133,23 +129,22 @@ private boolean isRecordInLatestReplay(final ConsumerRecord<byte[], byte[]> reco
...
 
     try {
       InternalTopicSerdes.deserializer(Command.class).deserialize(record.topic(), record.value());
     } catch (final Exception e) {
       throw new KsqlException(String.format(
           "Failed to backup record because it cannot deserialize value: %s",
-          new String(record.value(), StandardCharsets.UTF_8), e
-      ));
+          new String(record.value(), StandardCharsets.UTF_8)), e
+      );
```
Comment (big-andy-coates) [inline ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/CommandTopicBackupImpl.java:147]: Findbugs error no 2.

File: ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/CommandTopicBackupImpl.java [hunk 4]
```diff
@@ -193,18 +188,14 @@ public boolean commandTopicCorruption() {
...
   @VisibleForTesting
   BackupReplayFile openOrCreateReplayFile() {
-    final Optional<BackupReplayFile> latestFile = latestReplayFile();
-    if (latestFile.isPresent()) {
-      return latestFile.get();
-    }
-
-    return newReplayFile();
+    return latestReplayFile()
+        .orElseGet(this::newReplayFile);
```
Comment (big-andy-coates) [inline ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/CommandTopicBackupImpl.java:192]: simplified

File: ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/CommandTopicBackupImpl.java [hunk 5]
```diff
@@ -216,12 +207,11 @@ private BackupReplayFile newReplayFile() {
...
     if (files != null) {
       long latestTs = 0;
-      for (int i = 0; i < files.length; i++) {
-        final File bakFile = files[i];
+      for (final File bakFile : files) {
         final String bakTimestamp = bakFile.getName().substring(prefixFilename.length());
 
         try {
-          final Long ts = Long.valueOf(bakTimestamp);
+          final long ts = Long.parseLong(bakTimestamp);
```
Comment (big-andy-coates) [inline ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/CommandTopicBackupImpl.java:214]: warning fixed

File: ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/CommandTopicBackupImpl.java [hunk 6]
```diff
@@ -216,12 +207,11 @@ private BackupReplayFile newReplayFile() {
     File latestBakFile = null;
     if (files != null) {
       long latestTs = 0;
-      for (int i = 0; i < files.length; i++) {
-        final File bakFile = files[i];
+      for (final File bakFile : files) {
```
Comment (big-andy-coates) [inline ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/CommandTopicBackupImpl.java:210]: warning fixed

File: ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/CommandTopicBackupImpl.java [hunk 7]
```diff
@@ -230,17 +220,15 @@ private BackupReplayFile newReplayFile() {
...
-          continue;
         }
       }
     }
 
-    return (latestBakFile != null)
-        ? Optional.of(BackupReplayFile.writable(latestBakFile))
-        : Optional.empty();
+    return Optional.ofNullable(latestBakFile)
+        .map(BackupReplayFile::writable);
```
Comment (big-andy-coates) [inline ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/CommandTopicBackupImpl.java:228]: simpified

File: ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/restore/KsqlRestoreCommandTopic.java [hunk 8]
```diff
@@ -88,8 +86,9 @@ private static void throwOnInvalidRecords(final List<Pair<byte[], byte[]>> recor
       } catch (final Exception e) {
         throw new KsqlException(String.format(
             "Invalid CommandId string (line %d): %s",
-            n, new String(record.getLeft(), StandardCharsets.UTF_8), e
-        ));
+            n, new String(record.getLeft(), StandardCharsets.UTF_8)),
+            e
```
Comment (big-andy-coates) [inline ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/restore/KsqlRestoreCommandTopic.java:90]: Findbugs error no 3.

File: ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/restore/KsqlRestoreCommandTopic.java [hunk 9]
```diff
@@ -98,8 +97,9 @@ private static void throwOnInvalidRecords(final List<Pair<byte[], byte[]>> recor
       } catch (final Exception e) {
         throw new KsqlException(String.format(
             "Invalid Command string (line %d): %s",
-            n, new String(record.getRight(), StandardCharsets.UTF_8), e
-        ));
+            n, new String(record.getRight(), StandardCharsets.UTF_8)),
+            e
```
Comment (big-andy-coates) [inline ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/restore/KsqlRestoreCommandTopic.java:101]: Findbugs error no 4.

File: ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/restore/KsqlRestoreCommandTopic.java [hunk 10]
```diff
@@ -135,12 +135,7 @@ private static boolean promptQuestion() {
     final Console console = System.console();
     final String decision = console.readLine();
 
-    switch (decision.toLowerCase()) {
-      case "yes":
-        return true;
-      default:
-        return false;
-    }
+    return "yes".equals(decision.toLowerCase());
```
Comment (big-andy-coates) [inline ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/restore/KsqlRestoreCommandTopic.java:138]: simplified

**Review by spena — APPROVED** [review]
LGTM, thanks

**Pull request merged**